File: Tutoriels/Cpp/Patron_Conception/Creation/Singleton/src/GSingleton.cpp
//===============================================
#include "GSingleton.h"
//===============================================
GSingleton* GSingleton::m_instance = 0;
//===============================================
GSingleton::GSingleton() {
    m_data = "_NO_DATA_";
}
//===============================================
GSingleton::~GSingleton() {

}
//===============================================
GSingleton* GSingleton::Instance() {
    if(m_instance == 0) {
        m_instance = new GSingleton;
    }
    return m_instance;
}
//===============================================
void GSingleton::setData(const string& data) {
    m_data = data;
}
//===============================================
string GSingleton::getData() const {
    return m_data;
}
//===============================================
void GSingleton::showData() const {
    cout << m_data << "\n";
}
//===============================================

File: Tutoriels/CSharp/Patron_Conception/Creation/Singleton/src/GSingleton.cs
//===============================================
using System;
//===============================================
public class GSingleton {
    //===============================================
    private GSingleton() {
        Data = "_NO_DATA_";
    }
    //===============================================
    public static GSingleton Instance() {
        if(m_instance == null) {
            m_instance = new GSingleton();
        }
        return m_instance;
    }
    //===============================================
    public void Print() {
        Console.WriteLine(Data);
    }
    //===============================================
    public string Data {get; set;}
    //===============================================
    private static GSingleton m_instance = null;
    //===============================================
}
//===============================================
